backend/mem/fileSystem.go

```diff
@@ -0,0 +1,176 @@
+If a non-absolute path is given, an error is thrown.Additionally, a file does not
+technically exist until a call to "Touch()" is made on it. The "Touch" call links the
+file with the fileSystem's map and brings it into existence.
+If a file is written to before a touch call, Write() will take care of that call.  This is
+true for other functions as well and existence only poses a problem in the context of deletion
+or copying FROM a non-existent file.
+*/
+func (fs *FileSystem) NewFile(volume string, absFilePath string) (vfs.File, error) {
+
+	err := utils.ValidateAbsoluteFilePath(absFilePath)
+	if err != nil {
+		return nil, err
+	}
+
+	file, err := newFile(path.Base(absFilePath)) //validateAbsFile path will throw an error if there was a trailing slash, hence not calling path.Clean()
+	if err != nil {
+		return nil, err
+	}
+	tmp, err := fs.NewLocation(volume, utils.EnsureTrailingSlash(path.Dir(absFilePath)))
+	if err != nil {
+		return nil, err
+	}
+
+	file.location = tmp
+	return file, nil
+}
+
+/*
+NewLocation function returns the mem implementation of vfs.Location.
+A location does not exist unless at least one file lives on it. If a file
+is created on a non-existent location, then it will be created
+*/
+func (fs *FileSystem) NewLocation(volume string, absLocPath string) (vfs.Location, error) {
+
+	err := utils.ValidateAbsoluteLocationPath(absLocPath)
+	if err != nil {
+		return nil, err
+	}
+	str := utils.EnsureTrailingSlash(path.Clean(absLocPath))
+	return &Location{
+		fileSystem: fs,
+		name:       str,
+		exists:     false,
+		volume:     volume,
+	}, nil
+
+}
+
+// Name returns "mem"
+func (fs *FileSystem) Name() string {
+	return name
+}
+
+// Scheme return "file" as the initial part of a file URI ie: file://
+func (fs *FileSystem) Scheme() string {
+	return Scheme
+}
+func (fs *FileSystem) Initialize() {
+
+	fs.fsMap = make(map[string]objMap)
+
+}
+
+func init() {
+	backend.Register(Scheme, &FileSystem{make(map[string]objMap)})
+
+}
+
+//getKeys is used to get a list of absolute paths on a specified volume. These paths are a mixture of files and locations
+func (o objMap) getKeys() []string {
+	keyList := make([]string, 0)
+	for i := range o {
+		keyList = append(keyList, i)
+	}
+	return keyList
+}
+
+//fileHere returns a list of file pointers found at the absolute location path provided.  If none are there, returns an empty slice
+func (o objMap) filesHere(absLocPath string) []*File {
+
+	paths := o.getKeys()
+	fileList := make([]*File, 0)
+	for i := range paths {
+
+		object := o[paths[i]] //retrieve the object
+		if object.isFile {    //if the object is a file, cast its interface, i, to a file and append to the slice
+			file := object.i.(*File)
+			if file.Location().Path() == absLocPath {
+				fileList = append(fileList, file)
+			}
+		}
+	}
+	return fileList
+}
+
+//fileHere returns a list of file pointers found at the absolute location path provided.  If none are there, returns an empty slice
```

Comment:
wrong function name in the comment... actually the whole thing is copy/past of filesHere comment... should be updated with whatever fileNamesHere does.